tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictorService.cs

```diff
@@ -252,86 +229,62 @@ public CommandLineSuggestion GetSuggestion(PredictionContext context, int sugges
         }
 
         /// <inheritdoc/>
-        public virtual void RequestPredictions(IEnumerable<string> commands)
+        public virtual async Task<bool> RequestPredictionsAsync(IEnumerable<string> commands, CancellationToken cancellationToken)
         {
             Validation.CheckArgument(commands, $"{nameof(commands)} cannot be null.");
 
-            var localCommands= string.Join(AzPredictorConstants.CommandConcatenator, commands);
-            bool postSuccess = false;
-            Exception exception = null;
-            bool startRequestTask = false;
+            var localCommands = string.Join(AzPredictorConstants.CommandConcatenator, commands);
+            bool isRequestSent = false;
 
             try
             {
                 if (string.Equals(localCommands, _commandToRequestPrediction, StringComparison.Ordinal))
                 {
                     // It's the same history we've already requested the prediction for last time, skip it.
-                    return;
+                    return false;
                 }
 
                 if (commands.Any())
                 {
                     SetCommandToRequestPrediction(localCommands);
 
-                    // When it's called multiple times, we only need to keep the one for the latest command.
-
-                    _predictionRequestCancellationSource?.Cancel();
-                    _predictionRequestCancellationSource = new CancellationTokenSource();
+                    AzPredictorService.ReplaceThrottleUserIdToHeader(_client?.DefaultRequestHeaders, _azContext.UserId);
```

Comment:
Is there anyway we can hash the User ID? In the CLI we hash it since we only care about the uniquness for the limiting, not the actual ID. It may be fine in this case, I just don't have the context.

Reply:
The value is already hashed. I should rename the variable.